packages/store-sync/src/recs/syncToRecs.ts

```diff
@@ -59,9 +64,48 @@ export async function syncToRecs<TConfig extends StoreConfig = StoreConfig>({
 
   world.registerDisposer(stopSync);
 
+  const functionSelectorToResourceSelector = new Map<Hex, Hex>();
+
+  const getResourceSelector = async (functionSelector: Hex): Promise<Hex> => {
+    if (functionSelectorToResourceSelector.has(functionSelector)) {
+      return functionSelectorToResourceSelector.get(functionSelector) as Hex;
+    }
+
+    const entity = encodeEntity<KeySchema>(components.FunctionSelectors.metadata.keySchema, {
+      functionSelector,
+    });
+
+    const selectors = getComponentValue(components.FunctionSelectors, entity);
+
+    // If we can't find selectors due to not being synced yet, we can try to read them from the world contract
+    if (!selectors) {
+      // TODO make fieldLayout a table metadata field
+      const encodedFieldLayout = valueSchemaToFieldLayoutHex(components.FunctionSelectors.metadata.valueSchema);
+
+      const selectorRecord = (await publicClient.readContract({
+        address: address as Hex,
+        abi: IStoreAbi,
```

Comment:
this is a small thing that won't impact functionality, but given that function selectors exist as a world concept, might be clearer to use the `IBaseWorld` ABI here (it inherits `IStore` so it'll still have `getRecord`)